.ddev/providers/acquia.yaml

```diff
@@ -1,75 +1,71 @@
+     set -eu -o pipefail
+     if [ -z "${ACQUIA_API_KEY:-}" ] || [ -z "${ACQUIA_API_SECRET:-}" ]; then echo "Please make sure you have set ACQUIA_API_KEY and ACQUIA_API_SECRET in ~/.ddev/global_config.yaml" && exit 1; fi
+      if [ -z "${ACQUIA_ENVIRONMENT_ID:-}" ] ; then echo "Please set ACQUIA_ENVIRONMENT_ID via config.yaml or with '--environment=ACQUIA_ENVIRONMENT_ID=xxx'" && exit 1; fi
+      ssh-add -l >/dev/null || ( echo "Please 'ddev auth ssh' before running this command." && exit 1 )
+      acli -n auth:login -n --key="${ACQUIA_API_KEY}" --secret="${ACQUIA_API_SECRET}"
 
 db_pull_command:
   command: |
-    # set -x   # You can enable bash debugging output by uncommenting
     set -eu -o pipefail
+    # xargs here just trims whitespace
+    # We could use an easier technique when https://github.com/acquia/cli/issues/1629 is resolved
+    # just using `acli pull:db ${ACQUIA_ENVIRONMENT_ID}`
+    echo "Using ACQUIA_ENVIRONMENT_ID=${ACQUIA_ENVIRONMENT_ID}"
+    set -x   # You can enable bash debugging output by uncommenting
+    db_dump=$(acli pull:db ${ACQUIA_ENVIRONMENT_ID} --no-interaction --no-import | tail -2l | xargs)
     ls /var/www/html/.ddev >/dev/null # This just refreshes stale NFS if possible
-    pushd /var/www/html/.ddev/.downloads >/dev/null
-    drush @${project_id} sql-dump -n --alias-path=~/.drush --extra-dump=--no-tablespaces --gzip >db.sql.gz
+    cp ${db_dump} /var/www/html/.ddev/.downloads/db.sql.gz
 
-files_pull_command:
+files_import_command:
   command: |
     # set -x   # You can enable bash debugging output by uncommenting
     set -eu -o pipefail
-    ls /var/www/html/.ddev >/dev/null # This just refreshes stale NFS if possible
-    pushd /var/www/html/.ddev/.downloads >/dev/null;
-    drush -r docroot rsync --exclude-paths='styles:css:js' --alias-path=~/.drush -q -y @${project_id}:%files ./files
+    acli -n pull:files ${ACQUIA_ENVIRONMENT_ID}
 
 # push is a dangerous command. If not absolutely needed it's better to delete these lines.
-# db_push_command:
-#   command: |
-#     set -x   # You can enable bash debugging output by uncommenting
-#     set -eu -o pipefail
-#     TIMESTAMP=$(date +%y%m%d%H%M%S)
-#     ls /var/www/html/.ddev >/dev/null # This just refreshes stale NFS if possible
-#     cd /var/www/html/.ddev/.downloads
-#     drush rsync -y --alias-path=~/.drush ./db.sql.gz @${project_id}:/tmp/db.${TIMESTAMP}.sql.gz
-#     acli remote:ssh -n ${project_id} -- "cd /tmp && gunzip db.${TIMESTAMP}.sql.gz"
-#     acli remote:drush -n ${project_id} -- "sql-cli </tmp/db.${TIMESTAMP}.sql"
-#     acli remote:drush -n ${project_id} -- cr
-#     acli remote:ssh -n ${project_id} -- "rm /tmp/db.${TIMESTAMP}.*"
+db_push_command:
+  command: |
+    set -eu -o pipefail
+    export ACLI_DB_HOST=db ACLI_DB_NAME=db ACLI_DB_USER=db ACLI_DB_PASSWORD=db
+    set -x   # You can enable bash debugging output by uncommenting
+    acli push:db ${ACQUIA_ENVIRONMENT_ID} --no-interaction
```

Comment:
Noting the comment here. Probably we need to understand what this actually is before we make this change.